homeassistant/components/sensor/filesize.py

```diff
@@ -0,0 +1,94 @@
+"""
+Sensor for monitoring the size of a file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filesize/
+"""
+import datetime
+import logging
+import os
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.template import DATE_STR_FORMAT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_FILE_PATHS = 'file_paths'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FILE_PATHS): [cv.isfile],
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the file size sensor."""
+    sensors = []
+    for path in config.get(CONF_FILE_PATHS):
+        sensors.append(Filesize(path))
+
+    add_devices(sensors, True)
+
+
+class Filesize(Entity):
+    """Representation of the HA database."""
+
+    ICON = 'mdi:file'
+
+    def __init__(self, path):
+        """Initialize the data object."""
+        self._path = path   # Need to check its a valid path
+        self._size = None
+        self._last_updated = None
+        self._name = path.split("/")[-1]
+        self._unit_of_measurement = 'MB'
+        self.update()
+
+    def update(self):
+        """Get the size of the file."""
+        self._size = self.get_file_size(self._path)
+
+    def get_file_size(self, path):
+        """Returns the size of the file in MB."""
+        statinfo = os.stat(path)
+        decimals = 2
+        file_size = round(statinfo.st_size/1e6, decimals)
+        return file_size
+
+    def get_last_updated(self, path):
+        """Returns the time the file was last modified."""
+        statinfo = os.stat(path)
+        last_updated = datetime.datetime.fromtimestamp(statinfo.st_mtime)
+        last_updated = last_updated.strftime(DATE_STR_FORMAT)
+        return last_updated
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._size
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self.ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return other details about the sensor state."""
+        attrs = {}
+        attrs['path'] = self._path
+        attrs['last_updated'] = self.get_last_updated(self._path)
```

Comment:
Call `get_last_updated()` in `update()`.

Reply:
ok